### PR TITLE
Changed how InputValidation is implemented. Added material ui to Inpu…

### PR DESCRIPTION
**Pull request by joffe97**
…tValidation component. Added InputValidation to places it wasn't used.

Fix issue #131

* Changed how InputValidation is implemented. 
* Added material ui to InputValidation component. 
* Added InputValidation to places where validation were implemented without the InputValidation component.

### CODE REVIEW

File: GiEnJul/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (TheseusDeus) [inline GiEnJul/package-lock.json:1]: This file should no be in GiEnJul/*

File: GiEnJul/ClientApp/src/routes/registerGiver/ContactInfo.tsx [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import * as React from 'react';
 import { Button, Grid, Container } from '@material-ui/core';
 import useStyles from './Styles';
-import validator from 'validator';
```
Comment (TheseusDeus) [inline GiEnJul/ClientApp/src/routes/registerGiver/ContactInfo.tsx:4]: Have you checked that all of these changes wont interfere with the signup process and that it still connects to the API?

**Review by TheseusDeus — CHANGES_REQUESTED** [review]
At least remove the package-lock.json file

File: .gitignore [hunk 3]
```diff
@@ -348,3 +348,8 @@ MigrationBackup/
 
 # Ionide (cross platform F# VS Code tools) working folder
 .ionide/
+
+
+# Custom
+package.json
```
Comment (TheseusDeus) [inline .gitignore:354]: cant ignore these files in git. Need them in the /ClientApp/** folder for when implementing libraries

**Review by TheseusDeus — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (joffe97) [reply .gitignore]: Fixed this by adding `/` in front of the filename in .gitignore. This way it only ignores package.json and package-lock.json in the root folder. https://stackoverflow.com/questions/3637660/how-to-exclude-file-only-from-root-folder-in-git

Comment (TheseusDeus) [reply .gitignore]: Thats a great solution

**Review by TheseusDeus — APPROVED** [review]

**Pull request merged**